src/components/transport_manager/include/transport_manager/tcp/tcp_transport_adapter.h

```diff
@@ -49,9 +49,14 @@ class TcpTransportAdapter : public TransportAdapterImpl {
   /**
    * @brief Constructor.
    */
-  explicit TcpTransportAdapter(uint16_t port,
-                               resumption::LastState& last_state,
-                               const TransportManagerSettings& settings);
+  TcpTransportAdapter(uint16_t port,
+                      resumption::LastStateWrapperPtr last_state_wrapper,
+                      const TransportManagerSettings& settings);
+
+  DEPRECATED
+  TcpTransportAdapter(uint16_t port,
+                      resumption::LastState&,
+                      const TransportManagerSettings& settings) = delete;
```

Comment:
can't just do `= delete`, you need to define the function to work as it did before, otherwise this is a breaking change.

Reply:
@jacobkeeler Done in 4fdae6f97e8445b4aebe5269c4163e028115f508

Reply:
There is still no definition for this method

Reply:
@jacobkeeler Unfortunately, we can't work with the previous implementation of  `TcpTransportAdapter` constructor, because the previous implementation has been taking in a reference to a `LastState` instance as a parameter and initializing the  `TransportAdapterImpl` parent class by this reference. But in the current implementation the `TransportAdapterImpl` in constructor takes in the `LastStateWrapperPtr` entity as parameter , that is a `std::shared_ptr<LastStateWrapper>`. Also, I believe that your solution https://github.com/smartdevicelink/sdl_core/pull/2626#discussion_r372663000 won't work properly. In this case, `LastStateWrapper` during deleting his own instance will free the memory allocated by the original object (the memory of the ingoing instance, that has provided the reference for `TcpTransportAdapter` ). As a result, when the original object will try to delete his own instance memory corruption will occur (core crash).